src/features/projectsV2/ProjectsContext.tsx

```diff
@@ -271,7 +271,7 @@ export const ProjectsProvider: FC<ProjectsProviderProps> = ({
     projectSlug: string,
     siteIndex: number | null
   ) => {
-    if (!singleProject?.sites?.length) return;
+    if (!singleProject?.sites) return;
```

Comment:
I'm not sure why this was changed, but the sites array could be empty. In that case, we should not be able to set a selectedSite or add the site query param in the url